src/k2/cmd/ycsb/data.h

```diff
@@ -0,0 +1,155 @@
+/*
+MIT License
+
+Copyright(c) 2020 Futurewei Cloud
+
+    Permission is hereby granted,
+    free of charge, to any person obtaining a copy of this software and associated documentation files(the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and / or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions :
+
+    The above copyright notice and this permission notice shall be included in all copies
+    or
+    substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS",
+    WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+    DAMAGES OR OTHER
+    LIABILITY,
+    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+*/
+
+#pragma once
+
+#include <decimal/decimal>
+#include <string>
+
+#include <k2/common/Common.h>
+#include <k2/module/k23si/client/k23si_client.h>
+#include <k2/transport/Payload.h>
+#include <k2/transport/PayloadSerialization.h>
+
+#include "ycsb_rand.h"
+#include "Log.h"
+using namespace k2;
+static const String ycsbCollectionName = "YCSB";
+
+// helper function to convert to base (default 93 (ascii from 33 to 126)) representation
+String itostring(int n, int base = 93)
+{
+    String buf(String::initialized_later{}, 1000);
+    std::div_t dv{}; dv.quot = n;
+    uint32_t i = 0;
+
+    do {
+        dv = std::div(dv.quot, base);
+        buf[i] = (char)(std::abs(dv.rem)+33);  // avoid ascii value 126 because it is used for padding
+        i++;
+    } while(dv.quot);
+
+    return buf.substr(0,i);
+}
+
+class YCSBData{
+
+public:
+
+    static dto::Schema generate_schema(const uint32_t num_fields){
+
+        std::vector<dto::SchemaField> field_names;
+
+        for(uint32_t field_no = 0; field_no < num_fields; ++field_no){
+            String fieldname = "field" + std::to_string(field_no);
+            dto::SchemaField f {dto::FieldType::STRING, fieldname , false, false};
+            field_names.push_back(f);
+        }
+
+        dto::Schema ycsb_data_schema {
+            .name = "ycsb_data",
+            .version = 1,
+            .fields = field_names,
+            .partitionKeyFields = std::vector<uint32_t> { 0 },
+            .rangeKeyFields = std::vector<uint32_t> {}
+        };
+
+        return ycsb_data_schema;
+    }
+
+    static String idToKey(uint64_t keyid, uint32_t len_field){
+        String key = itostring(keyid);
+
+        if(key.size()>len_field)
+            return "";
```

Comment:
Should at least have a warning here, maybe throw an exception. Also style-wise you should include the brackets even for one line "if" statements.